packages/steps/src/BuildStepContext.ts

```diff
@@ -2,30 +2,53 @@ import os from 'os';
 import path from 'path';
 
 import { bunyan } from '@expo/logger';
+import { v4 as uuidv4 } from 'uuid';
 
 import { BuildStep } from './BuildStep.js';
 import { parseOutputPath } from './utils/template.js';
 import { BuildStepRuntimeError } from './errors.js';
 import { BuildRuntimePlatform } from './BuildRuntimePlatform.js';
+import { BuildStepEnv } from './BuildStepEnv.js';
 
-export class BuildStepContext {
+export interface BuildContextProvider {
+  readonly projectSourceDirectory: string;
+  readonly projectTargetDirectory: string;
+  readonly defaultWorkingDirectory: string;
+  readonly runtimePlatform: BuildRuntimePlatform;
+  readonly logger: bunyan;
+
+  readonly staticContext: any;
+
+  readonly env: BuildStepEnv;
+  updateEnv(env: BuildStepEnv): void;
+}
+
+export class BuildGlobalContext {
```

Comment:
I'm open to name suggestions. I considered also
- BuildStepGlobalContext
- BuildContext (it's no really clear when we use it in build-tools which build context is which)